### DISCUSSION TITLE
ch4-rpc/ch4-05-grpc-hack

### DISCUSSION THREAD

**Discussion by giscus** [chai2010/advanced-go-programming-book#613] [General]
# ch4-rpc/ch4-05-grpc-hack

Go语言高级编程

https://chai2010.cn/advanced-go-programming-book/ch4-rpc/ch4-05-grpc-hack.html

**Comment by ywsx0731** [comment]
golang 1.15之后不支持 x509了,需要使用SAN证书

**Comment by hBcDxG** [comment]
Token 认证里要求在grpc方法中调用`p.auth.Auth(ctx)`，流模式的grpc方法里没有`ctx`参数怎么做authentication？

**Reply by Jinvic** [reply]
流模式的ctx包装在steam内部，实现了`ServerStream`接口，通过`stream.Context()`获取。
